mantidproject/mantid
# Expose background subtraction options in GUI

Expose background subtraction options in GUI

Expose the background subtraction options in ReflectometryISISLoadAndProcess in the ISIS Reflectometry interface. The background processing instructions need to be specified on a per-angle basis so these are added to the Per-Angle Defaults table. The other options can be specified as defaults for all runs so are added as their own widgets on the Experiment Settings tab.

Fixes #27826

**To test:**

Requres the ISIS archive to be on.

- Open the `ISIS Reflectometry` interface
- Switch the instrument to `OFFSPEC` and enter run `53373` and angle `0.35` in the table.
- Click Process.
- Check the history of `IvsQ_binned_53373` and check that `SubtractBackground` is false and the other background inputs are default:
  - BackgroundCalculationMethod = PerDetectorAverage
  - DegreeOfPolynomial = 0
  - CostFunction = Least squares
- On the Experiment tab, tick SubtractBackground and then go back to the Runs tab and re-process. Check the history again and now SubtractBackground should be true with the others still defaults.
- Now change some of the other options and check their changes are reflected in the history.
- Also try entering processing instructions in the Per-angle defaults table, e.g. `380-389,416-426` and check that these appear in the history.
  - Try setting an angle in the Per-angle defaults table equal to `0.35` - it should be picked up in the history.
  - Try adding a new row with a different angle, e.g. `0.5` and set the Background. It should not be picked up in the history.
- Check that the background options can all be overriden by specifying properties manually in the Options field on the Runs table, e.g. `BackgroundProcessingInstructions='380-389,416-426'`
- From the Batch menu, save the settings to file and re-load them. Check they are all included when re-loaded.

**Release notes to follow**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
